

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Example walkthrough &mdash; Sex Genomics Toolkit 0.1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="RADSex-workflow" href="workflow.html" />
    <link rel="prev" title="Getting started" href="getting_started.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> Sex Genomics Toolkit
          

          
          </a>

          
            
            
              <div class="version">
                0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">RADSex</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Example walkthrough</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#preparing-the-data">Preparing the data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#generating-a-table-of-marker-depths-for-the-entire-dataset">Generating a table of marker depths for the entire dataset</a></li>
<li class="toctree-l2"><a class="reference internal" href="#computing-the-distribution-of-markers-between-sexes">Computing the distribution of markers between sexes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#finding-markers-significantly-associated-with-sex">Finding markers significantly associated with sex</a></li>
<li class="toctree-l2"><a class="reference internal" href="#aligning-markers-to-a-genome">Aligning markers to a genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="#going-further">Going further</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">RADSex-workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="input_files.html">Input files</a></li>
<li class="toctree-l1"><a class="reference internal" href="output_files.html">Output files</a></li>
</ul>
<p class="caption"><span class="caption-text">PSASS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../psass/introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/example.html">Example walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/workflow.html">PSASS-workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/output_files.html">Output files</a></li>
</ul>
<p class="caption"><span class="caption-text">KPOOL</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../kpool/introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/example.html">Example walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/workflow.html">Kpool-workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/output_files.html">Output files</a></li>
</ul>
<p class="caption"><span class="caption-text">sgtr</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../sgtr/introduction.html">Introduction</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Sex Genomics Toolkit</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Example walkthrough</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/radsex/example.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="example-walkthrough">
<h1>Example walkthrough<a class="headerlink" href="#example-walkthrough" title="Permalink to this headline">¶</a></h1>
<p>In this example, we will run RADSex on a public <em>Oryzias latipes</em> RAD-Sequencing dataset. We will detail each step of the process, highlight important details, and show how to use the R package <code class="docutils literal notranslate"><span class="pre">sgtr</span></code> to generate plots from the output of <code class="docutils literal notranslate"><span class="pre">radsex</span></code>. This guide assumes that <code class="docutils literal notranslate"><span class="pre">radsex</span></code> and the <code class="docutils literal notranslate"><span class="pre">sgtr</span></code> package have already been installed. For specific instruction about installing <code class="docutils literal notranslate"><span class="pre">radsex</span></code> and <code class="docutils literal notranslate"><span class="pre">sgtr</span></code>, check the <a class="reference internal" href="getting_started.html#install-release"><span class="std std-ref">Installation</span></a> section. All reported times and resources usage were measured on a desktop computer with an Intel i7-8700K 4.7 GHz processor, 32 Gb of memory, and a standard 7200 RPM Hard Disk Drive. Input data, results (with the exception of the markers depth table), and figures are provided in the <em>example</em> directory.</p>
<div class="section" id="preparing-the-data">
<h2>Preparing the data<a class="headerlink" href="#preparing-the-data" title="Permalink to this headline">¶</a></h2>
<p>The RAD-Sequencing datasets used* in this example are available on the Sequence Read Archive from NCBI. Reads were demultiplexed before being deposited on NCBI, and samples were grouped in two projects, males and females. The accession number for <strong>female</strong> samples is <strong>SRS662264</strong>, and the accession number for <strong>male</strong> samples is <strong>SRS662265</strong>. For convenience, simple scripts to download male and female samples from the EBI ftp can be found <a class="reference external" href="https://github.com/SexGenomicsToolkit/RADSex/tree/master/example/oryzias_latipes/data/download_female_samples.sh">here</a> for female samples and <a class="reference external" href="https://github.com/SexGenomicsToolkit/RADSex/tree/master/example/oryzias_latipes/data/download_male_samples.sh">here</a> for male samples. This dataset was published in <a class="reference external" href="http://www.genetics.org/content/early/2014/09/18/genetics.114.169284">Wilson et al 2014</a>.</p>
<p>A population map specifying the sex of each sample is provided <a class="reference external" href="https://github.com/SexGenomicsToolkit/RADSex/tree/master/example/oryzias_latipes/data/population_map.tsv">here</a>. The assembly used to align markers with <cite>radsex map</cite> was that of a HSOK strain, NCBI accession number <strong>GCA_002234695.1</strong> (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/assembly/GCA_002234695.1">link</a>). The chromosomes names file used to display chromosomes in genome plots is provided <a class="reference external" href="https://github.com/SexGenomicsToolkit/RADSex/tree/master/example/oryzias_latipes/data/chromosomes_names.tsv">here</a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>RADSex uses file names to generate individual IDs. Therefore, individual names in the population map must correspond to the file names without their extensions (<em>e.g.</em> the ID of an individual whose reads are in <strong>individual_1.fq.gz</strong> will be <strong>individual_1</strong>). Check the file names and population map provided above for an example of how to build the population map from file names. More details about the population map can be found in the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a> section.</p>
</div>
<p>From now on, we will assume the following directory structure:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>.
├─── samples
|     ├────── xxx.fastq.gz
|     ├────── xxx.fastq.gz
|     ├────── ...
|     └────── xxx.fastq.gz
├─── chromosomes_names.tsv
├─── genome.fasta
└─── popmap.tsv
</pre></div>
</div>
</div>
<div class="section" id="generating-a-table-of-marker-depths-for-the-entire-dataset">
<h2>Generating a table of marker depths for the entire dataset<a class="headerlink" href="#generating-a-table-of-marker-depths-for-the-entire-dataset" title="Permalink to this headline">¶</a></h2>
<p>The first step of RADSex is to create a table containing the depth of each marker in each individual for the entire dataset; a RADSex marker represents a non-polymorphic sequence (no mismatches or SNPs). This step is performed with the <code class="docutils literal notranslate"><span class="pre">process</span></code> command :</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsex</span> <span class="n">process</span> <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">dir</span> <span class="n">samples</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span> <span class="n">markers_table</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">8</span>
</pre></div>
</div>
<p><strong>Parameters</strong> (see the <a class="reference internal" href="usage.html#process-usage"><span class="std std-ref">process</span></a> usage section for details):</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--input-dir</span></code>: location of the demultiplexed reads directory. Supported reads file formats are described in the <a class="reference internal" href="input_files.html#reads-file"><span class="std std-ref">Reads files</span></a>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--output-file</span></code>: path to the markers depth table generated by this command.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--threads</span></code>: number of threads to use to process input files in parallel.</p></li>
</ul>
<p>The output file <em>markers_table.tsv</em> will be used as input for all analyses implemented in <code class="docutils literal notranslate"><span class="pre">radsex</span></code>, but it is not used for any <code class="docutils literal notranslate"><span class="pre">sgtr</span></code> plots. For more information about this file, check the <a class="reference internal" href="output_files.html#markers-depths-table-file"><span class="std std-ref">Markers depth table</span></a> section.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The parameter –min-depth specifies the minimum depth in at least one individual to retain a marker in the markers depth table. In most cases, we advise to keep the value of <code class="docutils literal notranslate"><span class="pre">--min-depth</span></code> to the default value <strong>1</strong> in order to retain all the information present in the dataset; this way, the markers depth table is only computed once, which is prefered as this step is by far the most computationally intensive in RADSex and markers can be filtered based on minimum depth in all downstream analyses. If you are certain that all individuals in your dataset were sequenced with high coverage and you do not plan to run analyses with a minimum depth of 1, you can specify a higher value for <code class="docutils literal notranslate"><span class="pre">--min-depth</span></code>.</p>
</div>
<p>With our setup, using 8 threads, this step completed in <strong>9 min 25 seconds</strong> with a peak memory usage of <strong>10.3 GB</strong>. The resulting markers depth table used 5.1 GB of disk space.</p>
</div>
<div class="section" id="computing-the-distribution-of-markers-between-sexes">
<h2>Computing the distribution of markers between sexes<a class="headerlink" href="#computing-the-distribution-of-markers-between-sexes" title="Permalink to this headline">¶</a></h2>
<p>The <code class="docutils literal notranslate"><span class="pre">distrib</span></code> command computes a table summarizing the distribution of all markers between males and females:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>radsex distrib --markers-table markers_table.tsv --output-file distribution.tsv --popmap-file popmap.tsv --min-depth 5``
</pre></div>
</div>
<p><strong>Parameters</strong> (see the <a class="reference internal" href="usage.html#distrib-usage"><span class="std std-ref">distrib</span></a> usage section for details):</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--markers-table</span></code>: path to the markers depth table generated in the previous step.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--output-file</span></code>: path to the distribution of markers between males and females generated by this command.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--popmap-file</span></code>: path to the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--min-depth</span></code>: minimum depth to consider a marker present in an individual.</p></li>
</ul>
<p>With our setup, this step completed in <strong>36 seconds</strong> with a peak memory usage of <strong>4 Mb</strong>.</p>
<p>The output file <em>distribution.tsv</em> is a tabulated file described in the <a class="reference internal" href="output_files.html#sex-distribution-file"><span class="std std-ref">Distribution of markers between groups</span></a> section. The distribution can be visualized with <code class="docutils literal notranslate"><span class="pre">sgtr</span></code> using the <code class="docutils literal notranslate"><span class="pre">radsex_distrib()</span></code> function:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsexvis</span><span class="p">::</span><span class="n">plot_sex_distribution</span><span class="p">(</span><span class="s2">&quot;distribution.tsv&quot;</span><span class="p">,</span> <span class="n">output_file_path</span> <span class="o">=</span> <span class="s2">&quot;distribution.png&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>To generate a basic plot, the only required parameter is the full path to a distribution table (<strong>“distribution.tsv”</strong> in this example). The figure can be directly saved to a file using the parameter <code class="docutils literal notranslate"><span class="pre">output_file_path</span></code>; if this parameter is not specified, the plot will be generated in the current R graphic device. For a full description of the <code class="docutils literal notranslate"><span class="pre">plot_sex_distribution()</span></code> function, including additional parameters, check the <a class="reference internal" href="../index.html#sgtr"><span class="std std-ref">sgtr</span></a> section.</p>
<p>The figure obtained with the previous command is displayed below:</p>
<img alt="../_images/distribution.png" src="../_images/distribution.png" />
<p>This figure is a tile plot with number of males on the x-axis and number of females on the y-axis. The color of a tile at coordinates (<strong>x</strong>, <strong>y</strong>) indicates the number of markers that were present in any <strong>x</strong> males and any <strong>y</strong> females. For instance, in this figure, there were between 25 and 99 markers found in 29 males (not necessarily always the same 29 males) and in 0 females. Tiles for which association with sex is significant (chi-squared test, using Bonferroni correction) are highlighted in red. Many markers found predominantly in males are significantly associated with sex, indicating that an XX/XY system determines sex in this species. Interestingly, there are no markers found in more than 29 out if 31 males and absent from all females, <em>i.e</em> no markers found at position (30, 0) and (31, 0). This suggests there may be male outliers in the dataset.</p>
<p>With our setup, this step completed in <strong>36 seconds</strong> with a peak memory usage of <strong>4 MB</strong>.</p>
</div>
<div class="section" id="finding-markers-significantly-associated-with-sex">
<h2>Finding markers significantly associated with sex<a class="headerlink" href="#finding-markers-significantly-associated-with-sex" title="Permalink to this headline">¶</a></h2>
<p>The <code class="docutils literal notranslate"><span class="pre">signif</span></code> command extracts all markers significantly associated sex from the dataset:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsex</span> <span class="n">signif</span> <span class="o">--</span><span class="n">markers</span><span class="o">-</span><span class="n">table</span> <span class="n">markers_table</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span> <span class="n">significant_markers</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">popmap</span><span class="o">-</span><span class="n">file</span> <span class="n">popmap</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">depth</span> <span class="mi">5</span>
</pre></div>
</div>
<p><strong>Parameters</strong> (see the <a class="reference internal" href="usage.html#signif-usage"><span class="std std-ref">signif</span></a> usage section for details):</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--markers-table</span></code>: path to the markers depth table generated with <code class="docutils literal notranslate"><span class="pre">process</span></code>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--output-file</span></code>: path to the markers depth table generated by this command. Markers can also be exported to a fasta file with the parameter <code class="docutils literal notranslate"><span class="pre">--output-fasta</span></code> (see the <a class="reference internal" href="output_files.html#fasta-file"><span class="std std-ref">Fasta file</span></a> section).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--popmap-file</span></code>: path to the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--min-depth</span></code>: minimum depth to consider a marker present in an individual.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The probability of association with sex is obtained with a chi-squared test on the number of females and males in which a marker is present. A marker is considered significantly associated with sex if its probability of association with sex is lower than 0.05 (this threshold can be adjusted with –signif-threshold) after Bonferroni correction. Markers significantly associated with sex are the ones in the tiles highlighted in red in the previous figure.</p>
</div>
<p>The markers depth table generated by <code class="docutils literal notranslate"><span class="pre">signif</span></code> can be visualized with <code class="docutils literal notranslate"><span class="pre">sgtr</span></code> using the <code class="docutils literal notranslate"><span class="pre">plot_depth()</span></code> function :</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsexvis</span><span class="p">::</span><span class="n">plot_coverage</span><span class="p">(</span><span class="s2">&quot;significant_markers.tsv&quot;</span><span class="p">,</span> <span class="n">output_file_path</span> <span class="o">=</span> <span class="s2">&quot;significant_markers.png&quot;</span><span class="p">,</span> <span class="n">popmap_file_path</span> <span class="o">=</span> <span class="s2">&quot;popmap.tsv&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>To generate a basic plot, the only required parameter is the full path to the subset of markers depth table (<strong>“significant_markers.tsv”</strong> in this example). The figure can be directly saved to a file using the parameter <code class="docutils literal notranslate"><span class="pre">output_file_path</span></code>; if this parameter is not specified, the plot will be generated in the current R graphic device. The parameter <code class="docutils literal notranslate"><span class="pre">popmap_file_path</span></code> can be specified to color individual IDs by sex. For a full description of the <code class="docutils literal notranslate"><span class="pre">plot_depth()</span></code> function, including additional parameters, check the <a class="reference internal" href="../index.html#sgtr"><span class="std std-ref">sgtr</span></a> section.</p>
<p>The resulting figure is displayed below:</p>
<img alt="../_images/significant_markers.png" src="../_images/significant_markers.png" />
<p>This figure is a heatmap with individuals on the x-axis and markers on the y-axis. The color of a tile at coordinates (<strong>x</strong>, <strong>y</strong>) indicates the depth of marker <strong>y</strong> in individual <strong>x</strong>. Both individuals and markers are clustered based on depth and clustering dendrograms are displayed by default. If a popmap is specified, male and female IDs are displayed with different colors.* In this example, two males are clustered with the females, confirming the results from <code class="docutils literal notranslate"><span class="pre">distrib</span></code> where male-specific markers were always missing from two males. These two males are actually genetic females whose sex was mis-assigned.</p>
<p>With our setup, this step completed in <strong>37 seconds</strong> with a peak memory usage of <strong>6 MB</strong>.</p>
</div>
<div class="section" id="aligning-markers-to-a-genome">
<h2>Aligning markers to a genome<a class="headerlink" href="#aligning-markers-to-a-genome" title="Permalink to this headline">¶</a></h2>
<p>When a reference genome is available, markers can be aligned to it in order to locate sex-differentiated regions. This is done using the <code class="docutils literal notranslate"><span class="pre">map</span></code> command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsex</span> <span class="nb">map</span> <span class="o">--</span><span class="n">markers</span><span class="o">-</span><span class="n">file</span> <span class="n">markers_table</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span> <span class="n">map_results</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">popmap</span><span class="o">-</span><span class="n">file</span> <span class="n">popmap</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">genome</span><span class="o">-</span><span class="n">file</span> <span class="n">genome</span><span class="o">.</span><span class="n">fasta</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">depth</span> <span class="mi">5</span>
</pre></div>
</div>
<p><strong>Parameters</strong> (see the <a class="reference internal" href="usage.html#map-usage"><span class="std std-ref">map</span></a> usage section for details):</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--markers-file</span></code>: path to a markers depth table generated with <code class="docutils literal notranslate"><span class="pre">process</span></code>, <code class="docutils literal notranslate"><span class="pre">distrib</span></code>, or <code class="docutils literal notranslate"><span class="pre">subset</span></code>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--output-file</span></code>: path to the alignment results table generated by this command.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--popmap-file</span></code>: path to the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--genome-file</span></code>: path to the genome file.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--min-depth</span></code>: minimum depth to consider a marker present in an individual.</p></li>
</ul>
<p>The output file <em>map_results.tsv</em> is a tabulated file described in the <a class="reference internal" href="output_files.html#mapping-results-file"><span class="std std-ref">Alignment results</span></a> section. This file can be visualized with <code class="docutils literal notranslate"><span class="pre">sgtr</span></code> using the <code class="docutils literal notranslate"><span class="pre">plot_genome()</span></code> function:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsexvis</span><span class="p">::</span><span class="n">plot_genome</span><span class="p">(</span><span class="s2">&quot;map_results.tsv&quot;</span><span class="p">,</span> <span class="n">chromosomes_names_file_path</span> <span class="o">=</span> <span class="s2">&quot;chromosomes_names.tsv&quot;</span><span class="p">,</span> <span class="n">output_file_path</span> <span class="o">=</span> <span class="s2">&quot;mapping_genome.png&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>To generate a basic plot, the only required parameters is the full path to the alignment results table (<strong>“mapping_results.tsv”</strong> in this example). The figure can be directly saved to a file using the parameter <code class="docutils literal notranslate"><span class="pre">output_file_path</span></code>; if this parameter is not specified, the plot will be generated in the current R graphic device. The parameter <code class="docutils literal notranslate"><span class="pre">chromosomes_names_file_path</span></code> can be used to specify chromosomes names as described in the <a class="reference internal" href="input_files.html#chromosomes-names"><span class="std std-ref">Chromosomes names file</span></a> section. For a full description of the <code class="docutils literal notranslate"><span class="pre">plot_genome()</span></code> function, including additional parameters, check the <a class="reference internal" href="../index.html#sgtr"><span class="std std-ref">sgtr</span></a> section.</p>
<p>The resulting figure is displayed below:</p>
<img alt="../_images/mapping_genome.png" src="../_images/mapping_genome.png" />
<p>This figure is a circular plot in which each sector corresponds to a chromosome and all unplaced scaffolds are grouped in an additional sector (not visible in this example as there are no unplaced scaffolds in this assembly). The top track gives the bias of a marker, 1 if the marker is present in all males and no females, and -1 if the marker is present in all females and no males. The bottom track shows the probability of association with sex (chi-squared test with Bonferroni correction).</p>
<p>Results for a specific region can be visualized with <code class="docutils literal notranslate"><span class="pre">sgtr</span></code> using the <code class="docutils literal notranslate"><span class="pre">plot_contig()</span></code> function:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsexvis</span><span class="p">::</span><span class="n">plot_contig</span><span class="p">(</span><span class="s2">&quot;mapping_results.tsv&quot;</span><span class="p">,</span> <span class="s2">&quot;genome.fasta.lengths&quot;</span><span class="p">,</span> <span class="s2">&quot;Chr01&quot;</span><span class="p">,</span> <span class="n">chromosomes_names_file_path</span> <span class="o">=</span> <span class="s2">&quot;chromosomes_names.tsv&quot;</span><span class="p">,</span> <span class="n">output_file_path</span> <span class="o">=</span> <span class="s2">&quot;mapping_contig.png&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>This function uses the same parameters as <code class="docutils literal notranslate"><span class="pre">plot_genome()</span></code> with the addition of the contig to plot (<em>Chr01</em> here). For a full description of the <code class="docutils literal notranslate"><span class="pre">plot_contig()</span></code> function, including additional parameters, check the <a class="reference internal" href="../index.html#sgtr"><span class="std std-ref">sgtr</span></a> section.</p>
<p>The resulting figure is displayed below:</p>
<img alt="../_images/mapping_contig.png" src="../_images/mapping_contig.png" />
<p>In this figure, bias and probability of association with sex defined just above are plotted against position on the plotted contig.</p>
<p>With our setup, this step completed in <strong>9 min 36 seconds</strong> with a peak memory usage of <strong>1.3 GB</strong>, most of the time being spent indexing the genome. If the genome is already indexed with BWA, this step completes in <strong>55 seconds</strong>.</p>
</div>
<div class="section" id="going-further">
<h2>Going further<a class="headerlink" href="#going-further" title="Permalink to this headline">¶</a></h2>
<p>In this example, we showed the most commonly used functions of <code class="docutils literal notranslate"><span class="pre">radsex</span></code> and <code class="docutils literal notranslate"><span class="pre">sgtr</span></code>, mostly using default parameters. In general, it is recommended to run <code class="docutils literal notranslate"><span class="pre">distrib</span></code> for several <code class="docutils literal notranslate"><span class="pre">min-depth</span></code> values (for instance 1, 2, 5, and 10) to better understand the distribution of marker depths in the dataset and estimate the robustness of markers significantly associated with sex. Three other commands are implemented in <code class="docutils literal notranslate"><span class="pre">radsex</span></code>:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">subset</span></code>: extract a subset of markers based on presence in number of males and females (see the <a class="reference internal" href="usage.html#subset-usage"><span class="std std-ref">subset</span></a> usage section)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">freq</span></code>: compute the distribution of presence of markers in all individuals from the dataset (see the <a class="reference internal" href="usage.html#freq-usage"><span class="std std-ref">freq</span></a> usage section)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">depth</span></code>: compute the minimum, maximum, median, and average marker depth for each individual from the dataset (see the <a class="reference internal" href="usage.html#depth-usage"><span class="std std-ref">depth</span></a> usage section)</p></li>
</ul>
<p>To get the full usage information for any <code class="docutils literal notranslate"><span class="pre">radsex</span></code> command, check the <a class="reference internal" href="usage.html#full-usage"><span class="std std-ref">Usage</span></a> section.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="workflow.html" class="btn btn-neutral float-right" title="RADSex-workflow" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="getting_started.html" class="btn btn-neutral float-left" title="Getting started" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018-2020, Romain Feron

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>