

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Input files &mdash; Sex Genomics Toolkit 0.1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Output files" href="output_files.html" />
    <link rel="prev" title="Usage" href="usage.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> Sex Genomics Toolkit
          

          
          </a>

          
            
            
              <div class="version">
                0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">RADSex</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="example.html">Example walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">RADSex-workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Input files</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#reads-files">Reads files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#population-map">Population map</a></li>
<li class="toctree-l2"><a class="reference internal" href="#chromosomes-names-file">Chromosomes names file</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="output_files.html">Output files</a></li>
</ul>
<p class="caption"><span class="caption-text">PSASS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../psass/introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/example.html">Example walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/workflow.html">PSASS-workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/output_files.html">Output files</a></li>
</ul>
<p class="caption"><span class="caption-text">KPOOL</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../kpool/introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/example.html">Example walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/workflow.html">Kpool-workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/output_files.html">Output files</a></li>
</ul>
<p class="caption"><span class="caption-text">sgtr</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../sgtr/introduction.html">Introduction</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Sex Genomics Toolkit</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Input files</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/radsex/input_files.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="input-files">
<h1>Input files<a class="headerlink" href="#input-files" title="Permalink to this headline">¶</a></h1>
<div class="section" id="reads-files">
<span id="reads-file"></span><h2>Reads files<a class="headerlink" href="#reads-files" title="Permalink to this headline">¶</a></h2>
<p>RADSex accepts demultiplexed reads files as input for <code class="docutils literal notranslate"><span class="pre">process</span></code>. RADSex should work with any demultiplexed RAD-sequencing reads files regardless of technology (single / double digest) or enzyme. RADSex cannot support paired-end reads because insert sizes are variable, and thus the second read in a pair does not cover a consistent region of the genome.</p>
<p>Input files can be in fasta or fastq formats and can be compressed with gzip. RADSex uses file extensions to detect input files and supports the following extensions: <strong>.fa</strong>, <strong>.fa.gz</strong>, <strong>.fq</strong>, <strong>.fq.gz</strong>, <strong>.fasta</strong>, <strong>.fasta.gz</strong>, <strong>.fastq</strong>, <strong>.fastq.gz</strong>, <strong>.fna</strong>, and <strong>.fna.gz</strong>.</p>
<p>Individual IDs are inferred from file names, <em>e.g.</em> RADSex will attribute the ID <strong>individual_1</strong> to the reads file <strong>individual1.fastq.gz</strong>.</p>
</div>
<div class="section" id="population-map">
<span id="id1"></span><h2>Population map<a class="headerlink" href="#population-map" title="Permalink to this headline">¶</a></h2>
<p>A population map file is a headerless TSV file (<em>i.e.</em> a tabulated file using “\t” - the “tab” character - as a separator) with individual ID in the first column and group in the second column. Groups can be any value and there can be more than two groups. However, most radsex analyses perform pairwise comparisons between groups and will require specifying the groups to compare with the <code class="docutils literal notranslate"><span class="pre">--groups</span></code> parameters (<em>e.g.</em> <code class="docutils literal notranslate"><span class="pre">--groups</span> <span class="pre">males,females</span></code>). If the popmap contains two groups, these groups will be used for pairwise group comparisons in the order they are found in the popmap (this order can be overridden with <code class="docutils literal notranslate"><span class="pre">--groups</span></code>).</p>
<p>An example of population map is given below:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">individual_1</span>    <span class="n">M</span>
<span class="n">individual_2</span>    <span class="n">M</span>
<span class="n">individual_3</span>    <span class="n">F</span>
<span class="n">individual_4</span>    <span class="n">N</span>
<span class="n">individual_5</span>    <span class="n">F</span>
</pre></div>
</div>
<p>Individual IDs can be any value, but it is important that they correspond to the name of the demultiplexed files (without the extension). For instance, the reads file for <em>individual_1</em> should be named <em>individual_1.fastq.gz</em> (or any fasta/fastq format supported by your demultiplexer).</p>
<p>If you are using Stacks with a barcodes file to demultiplexing the RAD-sequencing reads, just make sure that individual IDs in the barcodes file and in the population map are the same.</p>
</div>
<div class="section" id="chromosomes-names-file">
<span id="chromosomes-names"></span><h2>Chromosomes names file<a class="headerlink" href="#chromosomes-names-file" title="Permalink to this headline">¶</a></h2>
<p>Genome-wide results from the <code class="docutils literal notranslate"><span class="pre">map</span></code> command are visualized using the <code class="docutils literal notranslate"><span class="pre">plot_genome()</span></code> function of <code class="docutils literal notranslate"><span class="pre">radsex-vis</span></code>. This function automatically detects chromosomes if their name starts with “LG”, “CHR”, or “NC” (case unsensitive). If this is not the case, or if you would like to rename the chromosomes (for instance to shorten NCBI “NC_XXX” IDs), you should provide a chromosomes names file to <code class="docutils literal notranslate"><span class="pre">plot_genome()</span></code>.</p>
<p>The chromosomes names file is a headerless tabulated file with chromosome ID (from the genome file, <em>e.g.</em> “NC_XXXX”) in the first column and corresponding chromosome name in the second column (<em>e.g.</em> “Chrom1”).</p>
<p>An example of chromosomes names file is given below for the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/genome/?term=esox%20lucius">Northern Pike</a> genome.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">NC_025968</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG01</span>
<span class="n">NC_025969</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG02</span>
<span class="n">NC_025970</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG03</span>
<span class="n">NC_025971</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG04</span>
<span class="n">NC_025972</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG05</span>
<span class="n">NC_025973</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG06</span>
<span class="n">NC_025974</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG07</span>
<span class="n">NC_025975</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG08</span>
<span class="n">NC_025976</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG09</span>
<span class="n">NC_025977</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG10</span>
<span class="n">NC_025978</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG11</span>
<span class="n">NC_025979</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG12</span>
<span class="n">NC_025980</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG13</span>
<span class="n">NC_025981</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG14</span>
<span class="n">NC_025982</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG15</span>
<span class="n">NC_025983</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG16</span>
<span class="n">NC_025984</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG17</span>
<span class="n">NC_025985</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG18</span>
<span class="n">NC_025986</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG19</span>
<span class="n">NC_025987</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG20</span>
<span class="n">NC_025988</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG21</span>
<span class="n">NC_025989</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG22</span>
<span class="n">NC_025990</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG23</span>
<span class="n">NC_025991</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG24</span>
<span class="n">NC_025992</span><span class="o">.</span><span class="mi">3</span>     <span class="n">LG25</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Any scaffold included in the chromosomes names file will be considered a chromosome to be plotted as a sector. In most NCBI genomes, mitochondria are also named NC_XXX. As mitochondria are usually too small to be included as a sector in the circos plot, you should not add them to the chromosomes names file.</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="output_files.html" class="btn btn-neutral float-right" title="Output files" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="usage.html" class="btn btn-neutral float-left" title="Usage" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018-2020, Romain Feron

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>