

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Getting started &mdash; Sex Genomics Toolkit 0.1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Example walkthrough" href="example.html" />
    <link rel="prev" title="Introduction" href="introduction.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> Sex Genomics Toolkit
          

          
          </a>

          
            
            
              <div class="version">
                0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">RADSex</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Getting started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#installation">Installation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#requirements">Requirements</a></li>
<li class="toctree-l3"><a class="reference internal" href="#install-release">Installation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#update-radsex">Update RADSex</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#before-starting">Before starting</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-radsex">Running RADSex</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#computing-the-markers-depth-table">Computing the markers depth table</a></li>
<li class="toctree-l3"><a class="reference internal" href="#computing-the-distribution-of-markers-between-groups">Computing the distribution of markers between groups</a></li>
<li class="toctree-l3"><a class="reference internal" href="#extracting-markers-significantly-associated-with-sex">Extracting markers significantly associated with sex</a></li>
<li class="toctree-l3"><a class="reference internal" href="#aligning-markers-to-a-genome">Aligning markers to a genome</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="example.html">Example walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflow.html">RADSex-workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="input_files.html">Input files</a></li>
<li class="toctree-l1"><a class="reference internal" href="output_files.html">Output files</a></li>
</ul>
<p class="caption"><span class="caption-text">PSASS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../psass/introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/example.html">Example walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/workflow.html">PSASS-workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="../psass/output_files.html">Output files</a></li>
</ul>
<p class="caption"><span class="caption-text">KPOOL</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../kpool/introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/example.html">Example walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/workflow.html">Kpool-workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="../kpool/output_files.html">Output files</a></li>
</ul>
<p class="caption"><span class="caption-text">sgtr</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../sgtr/introduction.html">Introduction</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Sex Genomics Toolkit</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Getting started</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/radsex/getting_started.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="getting-started">
<h1>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
<div class="section" id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h2>
<div class="section" id="requirements">
<h3>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>A C++11 compliant compiler (GCC &gt;= 4.8.1, Clang &gt;= 3.3)</p></li>
<li><p>The zlib library (usually installed on linux by default)</p></li>
</ul>
</div>
<div class="section" id="install-release">
<span id="id1"></span><h3>Installation<a class="headerlink" href="#install-release" title="Permalink to this headline">¶</a></h3>
<p>There are three ways to install RADSex:</p>
<p><strong>1. Install the latest release</strong></p>
<ul class="simple">
<li><p>Download the latest release from <a class="reference external" href="https://github.com/SexGenomicsToolkit/RADSex/releases">GitHub</a></p></li>
<li><p>Unzip the archive</p></li>
<li><p>Navigate to the <cite>RADSex</cite> directory</p></li>
<li><p>Run <code class="docutils literal notranslate"><span class="pre">make</span></code></p></li>
</ul>
<p>The compiled <code class="docutils literal notranslate"><span class="pre">radsex</span></code> binary will be located in <strong>RADSex/bin/</strong>.</p>
<p><strong>2. Install the latest stable development version</strong></p>
<p>To install the latest stable version of RADSex directly from the GitHub repository, run the following commands:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">SexGenomicsToolkit</span><span class="o">/</span><span class="n">RADSex</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="n">RADSex</span>
<span class="n">make</span>
</pre></div>
</div>
<p>The compiled <code class="docutils literal notranslate"><span class="pre">radsex</span></code> binary will be located in <strong>RADSex/bin/</strong>.</p>
<p><strong>3. Install RADSex with conda</strong></p>
<p>RADSex is available in <a class="reference external" href="https://bioconda.github.io/recipes/radsex/README.html?#recipe-Recipe%20&amp;#x27;radsex&amp;#x27;">Bioconda</a>. To install RADSex with Conda, run the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">bioconda</span> <span class="n">radsex</span>
</pre></div>
</div>
</div>
<div class="section" id="update-radsex">
<h3>Update RADSex<a class="headerlink" href="#update-radsex" title="Permalink to this headline">¶</a></h3>
<p>To update RADSex, you can download the latest stable release and install it as described in the <a class="reference internal" href="#install-release"><span class="std std-ref">Installation</span></a> section.</p>
<p>If you installed RADSex directly from the GitHub repository, update RADSex by running the following commands from the <strong>RADSex</strong> directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">pull</span>
<span class="n">make</span> <span class="n">rebuild</span>
</pre></div>
</div>
<p>If you installed RADSex with Conda, run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">update</span> <span class="o">-</span><span class="n">c</span> <span class="n">bioconda</span> <span class="n">radsex</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="before-starting">
<h2>Before starting<a class="headerlink" href="#before-starting" title="Permalink to this headline">¶</a></h2>
<p>Before running the pipeline, you should prepare the following files:</p>
<ul class="simple">
<li><p>A <strong>set of demultiplexed reads</strong>. The current version of RADSex does not implement demultiplexing. Raw sequencing reads can be demultiplexed using <a class="reference external" href="http://catchenlab.life.illinois.edu/stacks/comp/process_radtags.php">Stacks</a> or <a class="reference external" href="http://nbviewer.jupyter.org/gist/dereneaton/af9548ea0e94bff99aa0/pyRAD_v.3.0.ipynb#The-seven-steps-described">pyRAD</a>.</p></li>
<li><p>A <strong>group information file (popmap)</strong>: a tabulated file with individual ID as the first column and group as the second column. It is important that the individual IDs in the popmap are the same as the names of the demultiplexed reads files (see the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a> section).</p></li>
<li><p>To align markers to a genome: the <strong>genome file</strong> in fasta format.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When visualizing <code class="docutils literal notranslate"><span class="pre">map</span></code> results with <code class="docutils literal notranslate"><span class="pre">sgtr</span></code>, linkage groups / chromosomes are automatically inferred from scaffold names in the reference sequence if their name starts with <em>LG</em>, <em>CHR</em>, or <em>NC</em> (case unsensitive). If chromosomes are named differently in the reference genome, you should prepare a tabulated file with reference contig ID in the first column and corresponding chromosome name in the second column (see the <a class="reference internal" href="input_files.html#chromosomes-names"><span class="std std-ref">Chromosomes names file</span></a>).</p>
</div>
</div>
<div class="section" id="running-radsex">
<h2>Running RADSex<a class="headerlink" href="#running-radsex" title="Permalink to this headline">¶</a></h2>
<div class="section" id="computing-the-markers-depth-table">
<span id="computing-depth-table"></span><h3>Computing the markers depth table<a class="headerlink" href="#computing-the-markers-depth-table" title="Permalink to this headline">¶</a></h3>
<p>The first step of RADSex is to create a table of marker depths for the entire dataset using the <code class="docutils literal notranslate"><span class="pre">process</span></code> command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsex</span> <span class="n">process</span> <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">dir</span> <span class="o">./</span><span class="n">samples</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span> <span class="n">markers_table</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">16</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">depth</span> <span class="mi">1</span>
</pre></div>
</div>
<p>In this example, demultiplexed reads are located in <strong>./samples</strong> and the markers table generated by <code class="docutils literal notranslate"><span class="pre">process</span></code> will be saved to <strong>markers_table.tsv</strong>. The parameter <code class="docutils literal notranslate"><span class="pre">--threads</span></code> specifies the number of threads to use, and <code class="docutils literal notranslate"><span class="pre">--min-depth</span></code> specifies the minimum depth to consider a marker present in an individual: markers which are not present with depth higher than this value in at least one individual will not be retained in the markers table.
It is advised to keep the minimum depth to the default value of 1 for this step, as it can be adjusted for each analysis later.</p>
<p>The resulting file <strong>markers_table.tsv</strong> is a tabulated file described in the <a class="reference internal" href="output_files.html#markers-depths-table-file"><span class="std std-ref">Markers depth table</span></a> section.</p>
</div>
<div class="section" id="computing-the-distribution-of-markers-between-groups">
<h3>Computing the distribution of markers between groups<a class="headerlink" href="#computing-the-distribution-of-markers-between-groups" title="Permalink to this headline">¶</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">distrib</span></code> command computes the distribution of markers between groups from a markers depth table:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>radsex distrib --markers-table markers_table.tsv --output-file distribution.tsv --popmap popmap.tsv --min-depth 5 --groups M,F``
</pre></div>
</div>
<p>In this example, <code class="docutils literal notranslate"><span class="pre">--markers-table</span></code> is the table generated in the <a class="reference internal" href="#computing-depth-table"><span class="std std-ref">Computing the markers depth table</span></a> section, and the distribution of markers between groups will be saved to <strong>distribution.tsv</strong>. The group of each individual in the population is given by <strong>popmap.tsv</strong> (see the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a> section). Groups of individuals to compare (as defined in the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a>) are specified manually with the parameter <code class="docutils literal notranslate"><span class="pre">--groups</span></code>. The minimum depth to consider a marker present in an individual is set to 5, meaning that markers with depth lower than 5 in an individual will not be considered present in this individual.</p>
<p>The resulting file <strong>distribution.tsv</strong> is a table described in the <a class="reference internal" href="output_files.html#sex-distribution-file"><span class="std std-ref">Distribution of markers between groups</span></a> section.</p>
<p>This distribution can be visualized with the <code class="docutils literal notranslate"><span class="pre">radsex_distrib()</span></code> function of <a class="reference external" href="https://github.com/SexGenomicsToolkit/sgtr">sgtr</a>, which generates a tile plot of marker counts with number of males on the x-axis and number of females on the y-axis.</p>
</div>
<div class="section" id="extracting-markers-significantly-associated-with-sex">
<h3>Extracting markers significantly associated with sex<a class="headerlink" href="#extracting-markers-significantly-associated-with-sex" title="Permalink to this headline">¶</a></h3>
<p>Markers significantly associated with sex are obtained with the <code class="docutils literal notranslate"><span class="pre">signif</span></code> command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsex</span> <span class="n">signif</span> <span class="o">--</span><span class="n">markers</span><span class="o">-</span><span class="n">table</span> <span class="n">markers_table</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span> <span class="n">markers</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">popmap</span> <span class="n">popmap</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">depth</span> <span class="mi">5</span> <span class="o">--</span><span class="n">groups</span> <span class="n">M</span><span class="p">,</span><span class="n">F</span> <span class="p">[</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">fasta</span> <span class="p">]</span>
</pre></div>
</div>
<p>In this example, <code class="docutils literal notranslate"><span class="pre">--markers-table</span></code> is the table generated in the <a class="reference internal" href="#computing-depth-table"><span class="std std-ref">Computing the markers depth table</span></a> section, and markers significantly associated with sex are saved to <strong>markers.tsv</strong>. The sex of each individual in the population is given by <strong>popmap.tsv</strong> (see the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a> section). Groups of individuals to compare (as defined in the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a>) are specified manually with the parameter <code class="docutils literal notranslate"><span class="pre">--groups</span></code>. The minimum depth to consider a marker present in an individual is set to 5, meaning that markers with depth lower than 5 in an individual will not be considered present in this individual.</p>
<p>By default, the <code class="docutils literal notranslate"><span class="pre">signif</span></code> function generates an output file in the same format as the markers depth table. Markers can also be exported to a fasta file using the <code class="docutils literal notranslate"><span class="pre">--output-fasta</span></code> parameter (see the <a class="reference internal" href="output_files.html#fasta-file"><span class="std std-ref">Fasta file</span></a> section).</p>
<p>The markers table generated by <code class="docutils literal notranslate"><span class="pre">signif</span></code> can be visualized with the <code class="docutils literal notranslate"><span class="pre">radsex_markers_depth()</span></code> function of <a class="reference external" href="https://github.com/SexGenomicsToolkit/sgtr">sgtr</a>, which generates a heatmap showing the depth of each marker in each individual.</p>
</div>
<div class="section" id="aligning-markers-to-a-genome">
<h3>Aligning markers to a genome<a class="headerlink" href="#aligning-markers-to-a-genome" title="Permalink to this headline">¶</a></h3>
<p>Markers can be aligned to a genome using the <code class="docutils literal notranslate"><span class="pre">map</span></code> command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsex</span> <span class="nb">map</span> <span class="o">--</span><span class="n">markers</span><span class="o">-</span><span class="n">file</span> <span class="n">markers_table</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span> <span class="n">alignment_results</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">popmap</span> <span class="n">popmap</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">genome</span><span class="o">-</span><span class="n">file</span> <span class="n">genome</span><span class="o">.</span><span class="n">fasta</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">quality</span> <span class="mi">20</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">frequency</span> <span class="mf">0.1</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">depth</span> <span class="mi">5</span> <span class="o">--</span><span class="n">groups</span> <span class="n">M</span><span class="p">,</span><span class="n">F</span>
</pre></div>
</div>
<p>In this example, <code class="docutils literal notranslate"><span class="pre">--markers-file</span></code> is the markers depth table generated in the <a class="reference internal" href="#computing-depth-table"><span class="std std-ref">Computing the markers depth table</span></a> step, and the path to the reference genome file is given by <code class="docutils literal notranslate"><span class="pre">--genome-file</span></code>; results will are saved to <strong>alignment_results.tsv</strong>. The sex of each individual in the population is given by <strong>popmap.tsv</strong> (see the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a> section), and the minimum depth to consider a marker present in an individual is set to 5, meaning that markers with depth lower than 5 in an individual will not be considered present in this individual. Groups of individuals to compare (as defined in the <a class="reference internal" href="input_files.html#population-map"><span class="std std-ref">Population map</span></a>) are specified manually with the parameter <code class="docutils literal notranslate"><span class="pre">--groups</span></code></p>
<p>The parameter <code class="docutils literal notranslate"><span class="pre">--min-quality</span></code> specifies the minimum mapping quality (as defined in <a class="reference external" href="http://bio-bwa.sourceforge.net/bwa.shtml">BWA</a>) to consider a marker properly aligned and is set to 20 in this example. The parameter <code class="docutils literal notranslate"><span class="pre">--min-frequency</span></code> specifies the minimum frequency of a marker in the population to retain this marker and is set to 0.1 here, meaning that only sequences present in at least 10% of individuals of the population are aligned to the genome.</p>
<p>The resulting file <code class="docutils literal notranslate"><span class="pre">mapping.tsv</span></code> is a table described in the <a class="reference internal" href="output_files.html#mapping-results-file"><span class="std std-ref">Alignment results</span></a> section.</p>
<p>Alignment results from <code class="docutils literal notranslate"><span class="pre">map</span></code> can be visualized with the <code class="docutils literal notranslate"><span class="pre">radsex_map_circos()</span></code> function of <a class="reference external" href="https://github.com/SexGenomicsToolkit/sgtr">sgtr</a>, which generates a circular plot showing bias and association with sex for each marker aligned to the genome. The same data can be shown in a manhattan plot with the <code class="docutils literal notranslate"><span class="pre">radsex_map_manhattan()</span></code> function.</p>
<p>Alignment results for a specific contig can be visualized with the <code class="docutils literal notranslate"><span class="pre">radsex_map_region()</span></code> function to show the same metrics for a single contig.</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="example.html" class="btn btn-neutral float-right" title="Example walkthrough" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="introduction.html" class="btn btn-neutral float-left" title="Introduction" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018-2020, Romain Feron

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>